src/Post_Command.php

```diff
@@ -1006,4 +999,22 @@ public function exists( $args ) {
 			WP_CLI::halt( 1 );
 		}
 	}
+
+	/**
+	 * Convert a date-time string with a hyphen separator to a space separator.
+	 *
+	 * @param string $date_string The date-time string to convert.
+	 * @return string The converted date-time string.
+	 *
+	 * Example:
+	 * maybe_convert_hyphenated_date_format( "2018-07-05-17:17:17" );
+	 * Returns: "2018-07-05 17:17:17"
+	 */
+	private function maybe_convert_hyphenated_date_format( $date_string ) {
```

Comment:
Is this a new behavior? Can you share more detail about it?

Reply:
No, it's actually done to ensure compatibility with previous versions. The [docs](https://developer.wordpress.org/cli/commands/post/generate/) advises using a datetime string with a hyphen between the date and time. If we don't remove this hyphen before calling `wp_insert_post`, the timestamp will be saved as 0.

Reply:
@rafaelzaleski Ok, great. Can you include some additional tests to capture this behavior then?

Also, it looks like your test introduces one break in a different test...

Reply:
> @rafaelzaleski Ok, great. Can you include some additional tests to capture this behavior then?

It's already [here](https://github.com/wp-cli/entity-command/blob/f6d53dfd0a8565b08f16a9633698b01170193cf6/features/post-generate.feature#L124) and [here](https://github.com/wp-cli/entity-command/blob/f6d53dfd0a8565b08f16a9633698b01170193cf6/features/post-generate.feature#L137).

> Also, it looks like your test introduces one break in a different test...

I'm not sure how to reproduce it locally. Do you have any ideas? 🤔

Reply:
> I'm not sure how to reproduce it locally. Do you have any ideas?

I tried to get tests running against PHP 5.6 locally and failed 😕

First, I tried this [act](https://github.com/nektos/act) tool. It was able to run the unit tests but not the functional tests:

![CleanShot 2023-09-08 at 07 29 06@2x](https://github.com/wp-cli/entity-command/assets/36432/bcb3c37a-a6f7-4be6-99d5-6255ce9b22a1)

I think this is the specific failure:

<img width="1735" alt="CleanShot 2023-09-08 at 07 29 36@2x" src="https://github.com/wp-cli/entity-command/assets/36432/473a8b54-1f6e-4aeb-81e4-cb2c642f94aa">

Next, I tried to install PHP 5.6 directly on my machine:

```
brew install shivammathur/php/php@5.6
brew link --overwrite --force shivammathur/php/php@5.6
```

Something is funky with the tests, though:

<img width="1771" alt="CleanShot 2023-09-08 at 07 31 33@2x" src="https://github.com/wp-cli/entity-command/assets/36432/3fa52f5c-b9ec-4d9b-b16e-38d2d9acb8bc">


I then manually tried to install WordPress 3.7, but received this MySQL error:

<img width="2324" alt="CleanShot 2023-09-08 at 07 34 56@2x" src="https://github.com/wp-cli/entity-command/assets/36432/a8816aa3-9bc5-4400-bc50-85b465c9629f">

My guess is that something changed in the behavior of `wp_insert_post()` between WP 3.7 and today.

Based on the failed test, it looks like your code changes the result of `--post_date="2018-07-01"`` from `2018-07-01 00:00:00` to `1970-01-01 00:00:00`

Reply:
Thank you @danielbachhuber. I think I was able to fix the edge cases on WP 3.7 and I have requested a new review.

Reply:
> I think I was able to fix the edge cases on WP 3.7

@rafaelzaleski Out of curiosity, how did you end up debugging/fixing this?

Reply:
> @rafaelzaleski Out of curiosity, how did you end up debugging/fixing this?

I encountered some challenges installing `3.7` locally too, mainly because of database issues.
 
So I went through the failed test data and reviewed the `wp_insert_post` source code from tag `3.7`. This helped me understand the nuances of the test data processing and the root cause of the failures. 
To ensure the solutions were working, I validated them using CI checks on my cloned repository.